src/common/dao/notification/notification_job.go

```diff
@@ -0,0 +1,141 @@
+package notification
+
+import (
+	"fmt"
+
+	"github.com/astaxie/beego/orm"
+	"github.com/goharbor/harbor/src/common/dao"
+	"github.com/goharbor/harbor/src/common/models"
+	"github.com/goharbor/harbor/src/common/utils/log"
+	"github.com/pkg/errors"
+)
+
+// UpdateNotificationJob update notification job
+func UpdateNotificationJob(job *models.NotificationJob, props ...string) (int64, error) {
+	if job == nil {
+		return 0, errors.New("nil job")
+	}
+
+	if job.ID == 0 {
+		return 0, fmt.Errorf("notification job ID is empty")
+	}
+
+	o := dao.GetOrmer()
+	return o.Update(job, props...)
+}
+
+// UpdateNotificationJobStatus update status of notification jobs whose status match conditions `statusCondition`,
+// if conditions provided but not matched, job status won't be update;
+// if conditions not provided, status of job with id will be update
+func UpdateNotificationJobStatus(id int64, status string, statusCondition ...string) (int64, error) {
```

Comment:
Per discussion this `UpdateNotificationJobStatus ` should be removed.

Reply:
> UpdateNotificationJobStatus

`UpdateNotificationJobStatus ` deleted